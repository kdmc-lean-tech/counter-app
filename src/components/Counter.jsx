import React, { useState } from 'react';
import PropTypes from 'prop-types';

const Counter = ({ value }) => {
  
  const defaultValue = value < 0 ? 0 : value > 100 ? 100 : value;
  const [count, setCount] = useState(defaultValue);

  const handleSum = () => {
    count < 100 && setCount(count + 1);
  }

  const handleSubstract = () => {
    count > 0 && setCount(count -1);
  }

  const handleReset = () => {
    setCount(0);
  }

  return (
    <div>
      <h1>Counter App</h1>
      <h2>Value: { count }</h2>
      <button onClick={ handleSum }>
        Increment
      </button>
      <button onClick={ handleSubstract }>
        Decrement
      </button>
      <button onClick={ handleReset }>
        Reset
      </button>
    </div>
  );
}

export default Counter;

Counter.propTypes = {
  value: PropTypes.number
}
