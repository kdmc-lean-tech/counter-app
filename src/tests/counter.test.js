import React from 'react';
import { shallow } from 'enzyme';
import Counter from '../components/Counter';

describe('Counter Test', () => {
  test('Should render <Counter/> succefully', () => {
    const wrapper = shallow(<Counter value={50} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should to show value by default is 50', () => {
    const wrapper = shallow(<Counter value={50} />);
    const textValue = wrapper.find('h2').text();
    expect(textValue).toEqual('Value: 50');
  });
  test('should to show value in 100 when prop is > 100', () => {
    const wrapper = shallow(<Counter value={101} />);
    const textValue = wrapper.find('h2').text();
    expect(textValue).toEqual('Value: 100');
  });
  test('should to show value in 0 when prop is < 0', () => {
    const wrapper = shallow(<Counter value={-1} />);
    const textValue = wrapper.find('h2').text().trim();
    expect(textValue).toEqual('Value: 0');
  });
  test('should to increment with button increment', () => {
    const wrapper = shallow(<Counter value={0} />);
    wrapper.find('button').at(0).simulate('click');
    const textValue = wrapper.find('h2').text().trim();
    expect(textValue).toEqual('Value: 1');
  });
  test('should to decrement with button decrement', () => {
    const wrapper = shallow(<Counter value={9} />);
    wrapper.find('button').at(1).simulate('click');
    const textValue = wrapper.find('h2').text().trim();
    expect(textValue).toEqual('Value: 8');
  });
  test('should to show value in 0 when to be click in decrement', () => {
    const wrapper = shallow(<Counter value={0} />);
    wrapper.find('button').at(1).simulate('click');
    const textValue = wrapper.find('h2').text().trim();
    expect(textValue).toEqual('Value: 0');
  });
  test('should to show value in 100 when to be click in increment', () => {
    const wrapper = shallow(<Counter value={100} />);
    wrapper.find('button').at(0).simulate('click');
    const textValue = wrapper.find('h2').text().trim();
    expect(textValue).toEqual('Value: 100');
  });
  test('should set value in 0 when to be click in reset', () => {
    const wrapper = shallow(<Counter value={50} />);
    wrapper.find('button').at(2).simulate('click');
    const textValue = wrapper.find('h2').text().trim();
    expect(textValue).toEqual('Value: 0');
  });
});
