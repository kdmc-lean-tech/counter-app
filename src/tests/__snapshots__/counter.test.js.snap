// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Test Should render <Counter/> succefully 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    Value: 
    50
  </h2>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;
